container/struct_args.go

```diff
@@ -0,0 +1,11 @@
+package container
+
+// StructArgs is a type which can be embedded in another struct to alert the
+// container that the fields of the struct are dependency inputs/outputs. That
+// is, the container will not look to resolve a value with StructArgs embedded
+// directly, but will instead use the struct's fields to resolve or populate
+// dependencies. Types with embedded StructArgs can be used in both the input
+// and output parameter positions.
+type StructArgs struct{}
```

Comment:
I thought it was simpler to have one struct vs `dig.In` and `dig.Out`... Wonder if anyone has an opinion

Reply:
In / Out makes it more clear when reading a structure definition, without going into the provider / usage:
* when I see a struct with`dig.In` I quickly associate it with a dependencies of some type. 
* when I see a struct with `dig.Out` I see that those are some dependencies a provider can provide.

Reply:
I'm happy to change it to In/Out, but note that I also have structs named Input and Output here to describe the actual constructor inputs/outputs... Any suggestions on naming?